### PR TITLE
Adds the label component; Small tweaks to the hand labeler

### PR DESCRIPTION
**Pull request by SteelSlayer**
## About The Pull Request
Adds the label component, which gets added to objects you label with the hand labeler. It is used to keep track of the label that was applied to it can be removed later on. This is paired with:

- You can now remove labels by hitting objects with the hand labeler while it's off.
- Limits the amount of labels an object can have to one. This was done mostly for code simplicity. I can probably make it possible to spam items with a bunch of labels again... if that's really wanted.

## Changelog
:cl:
code: Adds the label component, which gets added to objects you label with the hand labeler. It is used to keep track of the label that was applied to it can be removed later on.
soundadd: Adds a sound for labeling an object with the hand labeler
soundadd: Adds a sound for removing a label from an object with the hand labeler
tweak: Limits the amount of labels an object can have to one.
tweak: You can now remove labels from objects, by hitting it with the hand labeler, while it's off.
/:cl:

### CODE REVIEW

File: code/datums/components/label.dm [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	if(!isatom(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	label_name = _label_name
+
+	// Add the label to the name of the object in the format of: "Item name (label)"
+	var/atom/owner = parent
+	owner.name += " ([label_name])"
+
+	RegisterSignal(parent, COMSIG_PARENT_ATTACKBY, .proc/RemoveLabel)
```
Comment (optimumtact) [inline code/datums/components/label.dm]: You should do these in
http://codedocs.tgstation13.org/datum/component.html#proc/RegisterWithParent

So the component can be transferred

File: code/datums/components/label.dm [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/Examine)
+
+/datum/component/label/Destroy()
+	UnregisterSignal(parent, list(COMSIG_PARENT_ATTACKBY, COMSIG_PARENT_EXAMINE))
```
Comment (optimumtact) [inline code/datums/components/label.dm]: http://codedocs.tgstation13.org/datum/component.html#proc/UnregisterFromParent

File: code/datums/components/label.dm (hunk 1 above)
Comment (SteelSlayer) [reply code/datums/components/label.dm]: Alright, I'll do that, same with the unregisters

**Comment by cacogen** [comment]
these have been indelible always thank you. now that they can be removed can you make it so you can tag people?

**Comment by SteelSlayer** [comment]
It's possible but I'm pretty sure changing someone's name will cause it to show up over the radio. It would require some extra code to make the radio ignore the mob's label. I'll leave that up to someone else, if they want to add that feature.

File: code/datums/components/label.dm [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	var/obj/item/hand_labeler/labeler = attacker
+	if(!istype(labeler) || labeler.mode)
+		return
+
+	var/atom/owner = source // Source will be the owner / parent of this component.
+	owner.name = replacetext(owner.name, "([label_name])", "") // Remove the label text from the parent's name, wherever it may be.
+	owner.name = trim(owner.name) // Shave off any white space from the beginning or end of the parent's name.
+	playsound(owner, 'sound/items/poster_ripped.ogg', 20, TRUE)
+	to_chat(user, "<span class='warning'>You remove the label from [owner].</span>")
+	Destroy() // Remove the component from the object.
```
Comment (Cyberboss) [inline code/datums/components/label.dm]:
```suggestion
	qdel(src)
```

File: code/datums/components/label.dm [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	owner.name += " ([label_name])"
+
+/datum/component/label/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_PARENT_ATTACKBY, .proc/RemoveLabel)
+	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/Examine)
+
+/datum/component/label/UnregisterFromParent()
+	UnregisterSignal(parent, list(COMSIG_PARENT_ATTACKBY, COMSIG_PARENT_EXAMINE))
+
+/datum/component/label/proc/RemoveLabel(datum/source, obj/item/attacker, mob/user, params)
```
Comment (Cyberboss) [inline code/datums/components/label.dm]: Missing autodoc for the proc

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/datums/components/label.dm [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+		return
+
+	var/atom/owner = source // Source will be the owner / parent of this component.
+	owner.name = replacetext(owner.name, "([label_name])", "") // Remove the label text from the parent's name, wherever it may be.
+	owner.name = trim(owner.name) // Shave off any white space from the beginning or end of the parent's name.
+	playsound(owner, 'sound/items/poster_ripped.ogg', 20, TRUE)
+	to_chat(user, "<span class='warning'>You remove the label from [owner].</span>")
+	Destroy() // Remove the component from the object.
+
+/datum/component/label/proc/Examine(datum/source, mob/user, list/examine_list)
```
Comment (Cyberboss) [inline code/datums/components/label.dm]: Missing autodoc

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by cacogen** [comment]
>I'm pretty sure changing someone's name will cause it to show up over the radio

isn't your voice and your atom name handled by two different vars

**Comment by SteelSlayer** [comment]
@cacogen I couldn't tell you to be honest. That's very cool if it is.

**Comment by cacogen** [comment]
It is cool. It is.

**Review by spookydonut — APPROVED** [review]

**Pull request merged**